src/main/java/org/opensearch/neuralsearch/processor/chunker/ChunkerFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.Locale;
+import java.util.function.Function;
+
+/**
+ * A factory to create different chunking algorithm objects and return all supported chunking algorithms.
+ */
+public class ChunkerFactory {
+
+    private static final ImmutableMap<String, Function<Map<String, Object>, Chunker>> chunkers = ImmutableMap.of(
+        FixedTokenLengthChunker.ALGORITHM_NAME,
+        FixedTokenLengthChunker::new,
+        DelimiterChunker.ALGORITHM_NAME,
+        DelimiterChunker::new
+    );
+
+    public static Chunker create(final String type, final Map<String, Object> parameters) {
+        Function<Map<String, Object>, Chunker> chunkerConstructionFunction = chunkers.get(type);
+        if (chunkerConstructionFunction == null) {
```

Comment:
The map in constructed in our own code, so when the key is correct we can definitely get a non null value, if we already validated the input, this check can be avoid, right?

Reply:
For the create method in `ChunkerFactory`, the algorithm name has not been validated.